### PR TITLE
Dlg reorder des

### PR DESCRIPTION
**Pull request by lilyclements**
A few points with this - 
I'm not sure how to do .SetParameter or .ControlContentsChanged for ucrReorderObjects?

Additionally, there is a bug which I have outlined in #2590 
This bug was pre-existing before any code changes I made to the dialog

UPDATE: This is ready for Review

### CODE REVIEW

**Comment by Lunalo** [comment]
Hi @lilyclements 

I think you can do inherits from ucrCore in the designer code 
image

image

**Comment by lilyclements** [comment]
Great thanks! I believe this has been done so I just need to wait for #2606 to be merged.

File: instat/dlgReoderDescriptives.vb [hunk 1]
```diff
@@ -16,54 +16,66 @@
...
+        ' ucrSelector DataFrame
+        ucrDataFrameReorder.SetParameter(New RParameter("data_name", 0))
+        ucrDataFrameReorder.SetParameterIsString()
+
+        ' ucrReorderObjects
+        '        ucrReorderObjects.SetParameter(New RParameter("new_order", 1))
+        ' ucrReorderObjects.SetParameterIsString()
+
+        'clsDefaultFunction.AddParameter("new_order", ucrReorderObjects.GetVariableNames)
+
```
Comment (stevenndungu) [inline instat/dlgReoderDescriptives.vb]: This should be set on the setdefault sub

File: instat/dlgReoderDescriptives.vb [hunk 2]
```diff
@@ -16,54 +16,66 @@
...
+        ucrReorderObjects.setDataType("object")
+        ucrReorderObjects.setDataframes(ucrDataFrameReorder)
     End Sub
+
     Private Sub ReopenDialog()
 
     End Sub
+
```
Comment (stevenndungu) [inline instat/dlgReoderDescriptives.vb]: This sub is not called anywhere, and thus should be removed. Plus it does not handle any event on the code.

**Comment by stevenndungu** [comment]
otherwise the dialogue looks good for Danny

**Comment by dannyparsons** [comment]
@stevenndungu So it's not ready for me yet?

**Comment by stevenndungu** [comment]
It is ready, apart from the above mentioned two things, the dialogue is working fine, as it was previously.

**Comment by dannyparsons** [comment]
If there are still corrections to be made then it's not ready yet.
I think it will be clearer for everyone if the reviewer waits until all suggested changes to be made and then they can say it's ready.

**Comment by lilyclements** [comment]
@africanmathsinitiative/developers This is ready for review

**Comment by lilyclements** [comment]
@africanmathsinitiative/developers This is still ready for review

File: instat/dlgReoderDescriptives.vb [hunk 3]
```diff
@@ -13,57 +13,67 @@
...
+        ' ucrSelector DataFrame
+        ucrDataFrameReorder.SetParameter(New RParameter("data_name", 0))
+        ucrDataFrameReorder.SetParameterIsString()
+
+        ' ucrReorderObjects
+        ucrReorderObjects.SetParameter(New RParameter("new_order", 1))
+
+        ucrReorderObjects.setDataType("object")
+        ucrReorderObjects.setDataframes(ucrDataFrameReorder)
     End Sub
```
Comment (lilyclements) [inline instat/dlgReoderDescriptives.vb]: .

**Comment by lilyclements** [comment]
@africanmathsinitiative/developers This has been waiting for review for some time over the two-day-period we decided on

**Comment by lilyclements** [comment]
@africanmathsinitiative/developers This is still ready (amongst others)! I'll be there in a few days so there's no escaping my pull requests for much longer... :p

**Comment by stevenndungu** [comment]
Everything is perfect! @dannyparsons  this is ready for you to review.

**Comment by Lunalo** [comment]
@dannyparsons 

I have tested this and it is ready for merging

**Pull request merged**